hbase-backup/src/test/java/org/apache/hadoop/hbase/backup/TestFullRestore.java

```diff
@@ -71,6 +72,44 @@ public void testFullRestoreSingle() throws Exception {
     hba.close();
   }
 
+  @Test
+  public void testFullRestoreSingleWithRegion() throws Exception {
+    LOG.info("test full restore on a single table empty table that has a region");
+
+    // This test creates its own table so other tests are not affected (we adjust it in this test)
+    TableName tableName = TableName.valueOf("table-full-restore-single-region");
+    TEST_UTIL.createTable(tableName, famName);
+
+    Admin admin = TEST_UTIL.getAdmin();
+
+    // Add & remove data to ensure a region is active, but functionally empty
+    Table table = TEST_UTIL.getConnection().getTable(tableName);
+    loadTable(table);
+    admin.flush(tableName);
+    TEST_UTIL.deleteTableData(tableName);
+    admin.flush(tableName);
+
+    admin.majorCompact(tableName);
+    Thread.sleep(5_000);
```

Comment:
This could end up being flaky if the compaction doesn't finish within 5s. It might be better to use TEST_UTIL.compact which blocks

Reply:
Thanks, didn't know TEST_UTIL supported this.